include/libopencm3/swm050/flash.h

```diff
@@ -0,0 +1,42 @@
+/** @defgroup flash_defines Flash Defines
+ *
+ * @brief <b>Defined Constants and Types for the SWM050 Flash API</b>
+ *
+ * @ingroup SWM050_defines
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2019 Caleb Szalacinski <contact@skiboy.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**@{*/
+#ifndef LIBOPENCM3_FLASH_H
+#define LIBOPENCM3_FLASH_H
+
+#include <libopencm3/cm3/common.h>
+
+BEGIN_DECLS
+
+uint32_t flash_write(uint32_t *dest, uint32_t *src, uint8_t cnt);
+uint32_t flash_read(uint32_t *src, uint32_t *dest, uint8_t cnt);
```

Comment:
is this a swm050 pecularity that flash isn't memory mapped?

Reply:
Yes.  I'm fairly certain it is a separate peripheral.  I tried some writes to flash that replied with the correct status code, but they didn't show up on a dump of the main flash.  If it is memory-mapped, it is not documented where.

Reply:
if I understand from the later .c file (and the datasheet), this is not "flash" in the sense of the "flash.c" functions in any other families.  This is for a special  0.5k "user flash" region, separate from the normal data section.  I'm not sure what call it, and I know synwit just calls it "data flash" vs "program flash" so *shrugs* but it seemed confusing.